src/org/anjocaido/groupmanager/dataholder/WorldDataHolder.java

```diff
@@ -510,10 +503,8 @@ public void removeUsersChangedFlag() {
 	public void removeGroupsChangedFlag() {
 
 		setGroupsChanged(false);
-		synchronized (groups) {
-			for (Group g : getGroups().values()) {
-				g.flagAsSaved();
-			}

```

Comment:
I don't have an environment where I can do prolonged and extensive testing, so either this would need to be reverted or it would be published as a potentially highly unstable build